### PR TITLE
Fix button on Choose Security Method page

### PR DESCRIPTION
**Pull request by marcinbodnar**

### CODE REVIEW

File: src/utils/wallet.js [hunk 1]
```diff
@@ -689,7 +689,16 @@ class Wallet {
     async setupRecoveryMessageNewAccount(accountId, method, securityCode, fundingOptions, recoverySeedPhrase) {
         const { secretKey } = parseSeedPhrase(recoverySeedPhrase)
         const recoveryKeyPair = KeyPair.fromString(secretKey)
-        await this.validateSecurityCode(accountId, method, securityCode);
+        
+        try {
+            await this.validateSecurityCode(accountId, method, securityCode);
+        } catch (error) {
+            if (error.toString().includes('Error: Invalid code')) {
+                throw new WalletError(error, 'setupRecoveryMessageNewAccount.invalidCode')
```
Comment (vgrichina) [inline src/utils/wallet.js:697]: Isn’t first parameter supposed to be a string vs error?

File: src/utils/wallet.js [hunk 2]
```diff
@@ -689,7 +689,16 @@ class Wallet {
     async setupRecoveryMessageNewAccount(accountId, method, securityCode, fundingOptions, recoverySeedPhrase) {
         const { secretKey } = parseSeedPhrase(recoverySeedPhrase)
         const recoveryKeyPair = KeyPair.fromString(secretKey)
-        await this.validateSecurityCode(accountId, method, securityCode);
+        
+        try {
+            await this.validateSecurityCode(accountId, method, securityCode);
+        } catch (error) {
+            if (error.toString().includes('Error: Invalid code')) {
```
Comment (vgrichina) [inline src/utils/wallet.js:696]: Where is this error coming from? Latest near-api-js should have type field in errors.

File: src/utils/wallet.js (hunk 1 above)
Comment (marcinbodnar) [reply src/utils/wallet.js]: @vgrichina I refactored it, `validateSecurityCode` was already handling this error, just an alert was not showing

File: src/utils/wallet.js (hunk 2 above)
Comment (marcinbodnar) [reply src/utils/wallet.js]: @vgrichina It's from CH, no `type`:
https://github.com/near/near-wallet/blob/master/src/utils/wallet.js#L665

File: src/utils/wallet.js [hunk 3]
```diff
@@ -667,7 +667,7 @@ class Wallet {
                 await this.postSignedJson('/account/validateSecurityCode', body);
             }
         } catch(e) {
-            throw new WalletError('Invalid code', 'setupRecoveryMessage.error')
+            throw new WalletError('Invalid code', 'setupRecoveryMessageNewAccount.invalidCode')
```
Comment (vgrichina) [inline src/utils/wallet.js:670]: @marcinbodnar let's have a check for specific error from contract helper. I believe it provides 401 code + messages on what exactly happened:
https://github.com/near/near-contract-helper/blob/f16b2f8d5d691c4174e5ee33636005f735775220/middleware/2fa.js#L243

Comment (marcinbodnar) [reply src/utils/wallet.js:670]: @vgrichina this error is for email and phone recovery code, so it's this one in CH:
https://github.com/near/near-contract-helper/blob/f16b2f8d5d691c4174e5ee33636005f735775220/app.js#L291

But it throws `recoveryMethod does not exist` which doesn't seem to be relevant. Want me to check this message string anyway?

Comment (vgrichina) [reply src/utils/wallet.js:670]: @marcinbodnar ah, got it

it is relatively relevant, think of it same as when validating user/password you usually just report that there is no user with such password (vs revealing that user exists)

not sure how important though, maybe we should give more granular info

but I think we need to catch this specific error if we are only expecting it

Comment (vgrichina) [reply src/utils/wallet.js:670]: @marcinbodnar anyway, gonna merge – let's fix in follow-up.

Comment (marcinbodnar) [reply src/utils/wallet.js:670]: @vgrichina 
> it is relatively relevant, think of it same as when validating user/password you usually just report that there is no user with such password (vs revealing that user exists)

Right, got it. I will add this check in my next PR.

**Pull request merged**